nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -193,6 +253,90 @@ Route GoalIntentExtractor::pruneStartandGoal(
   return pruned_route;
 }
 
+unsigned int GoalIntentExtractor::associatePoseWithGraphNode(
+  std::vector<unsigned int> node_indices,
+  const geometry_msgs::msg::PoseStamped & pose)
+{
+  unsigned int s_mx, s_my, g_mx, g_my;
+  if (!costmap_sub_->getCostmap()->worldToMap(
+      pose.pose.position.x, pose.pose.position.y,
+      s_mx, s_my))
+  {
+    throw nav2_core::StartOutsideMapBounds(
+            "Start Coordinates of(" + std::to_string(pose.pose.position.x) + ", " +
+            std::to_string(pose.pose.position.y) + ") was outside bounds");
+  }
+
+  if (costmap_sub_->getCostmap()->getCost(s_mx, s_my) == nav2_costmap_2d::LETHAL_OBSTACLE) {
+    throw nav2_core::StartOccupied("Start was in lethal cost");
+  }
+
+  //TODO(jwallace42): double check the frames, probably need to move into the map frame...
```

Comment:
TODO? I think this is all correct, no?

Reply:
Well, 

the route graph could be in map frame and the local costmap could be in odom frame.

Reply:
True, if using the local costmap this could check if the `costmap->getGobalFrame()` is the same as the route serverframe. If not, do that transform